File: testproject-client/src/main/java/com/testproject/app/cliente/service/ClienteService.java
package com.testproject.app.cliente.service;

import java.util.List;

import com.testproject.app.cliente.entity.Cliente;

public interface ClienteService {
	public List<Cliente> getClientes();
	public Cliente getCliente(Long id);
	public Cliente getCliente(String cedula);
	public Cliente insertCliente(Cliente cliente);
	public Cliente updateCliente(Cliente cliente);
	public Cliente deleteCliente(Long id);	
	public List<Cliente> getClientes(String apellidoPaterno, String apellidoMaterno);
}

File: testproject-web/services/cliente.Service.js
import fetchWrapper from '../helpers/fetch-wrapper';
const API_ENDPOINT = process.env.NEXT_PUBLIC_API_ENDPOINT;
//const api = "http://localhost:8092"
//const api = "http://host.docker.internal:8092"
const api = "http://host.docker.internal:9090/cliente-ws"
const APICLIENTE = `${api}/clientes`;

const clienteService = {
    getClientes: async() => {        
        return await (fetchWrapper.get(`${APICLIENTE}`));
    },
    getClientePorCedula: async(cedula) => {
        return await (fetchWrapper.get(`${APICLIENTE}/clientebycedula?cedula=${cedula}`));
    },
    insertCliente: async(data) => {        
        return await (fetchWrapper.post(`${APICLIENTE}`, data));
    },
    updateCliente: async(data) => {
        return await (fetchWrapper.put(`${APICLIENTE}`, data));
    },
    deleteCliente: async(id) => {
        return await (fetchWrapper.delete(`${APICLIENTE}/${id}`));
    }
}

export default clienteService;